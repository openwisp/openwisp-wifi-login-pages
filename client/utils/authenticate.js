import {sessionStorage} from "./storage";

const authenticate = (cookies, orgSlug) => {
  const token = cookies.get(`${orgSlug}_auth_token`);
  const sessionKey = sessionStorage.getItem(`${orgSlug}_auth_token`);
  if (sessionKey) {
    cookies.remove(`${orgSlug}_auth_token`, {path: "/"});
    cookies.remove(`${orgSlug}_username`, {path: "/"});
    return true;
  }
  if (token) return true;
  return false;
};
export default authenticate;
