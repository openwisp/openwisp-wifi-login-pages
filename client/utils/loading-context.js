import React from "react";

export const loadingContextValue = {setLoading: () => {}, getLoading: () => {}};
const LoadingContext = React.createContext(loadingContextValue);

export default LoadingContext;
