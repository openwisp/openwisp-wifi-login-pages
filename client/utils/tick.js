const tick = function () {
  return new Promise((resolve) => {
    process.nextTick(resolve);
  });
};

export default tick;
