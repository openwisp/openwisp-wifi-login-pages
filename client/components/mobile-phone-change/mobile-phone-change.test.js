/* eslint-disable prefer-promise-reject-errors */
/* eslint-disable camelcase */
import axios from "axios";
import {mount, shallow} from "enzyme";
import React from "react";
import PropTypes from "prop-types";
import {Cookies} from "react-cookie";
import {toast} from "react-toastify";
import {Provider} from "react-redux";
import {Redirect, Router} from "react-router-dom";
import {createMemoryHistory} from "history";
import {loadingContextValue} from "../../utils/loading-context";
import loadTranslation from "../../utils/load-translation";
import getConfig from "../../utils/get-config";
import tick from "../../utils/tick";
import MobilePhoneChangeWrapped from "./mobile-phone-change";
import validateToken from "../../utils/validate-token";
import submitOnEnter from "../../utils/submit-on-enter";

const MobilePhoneChange = MobilePhoneChangeWrapped.WrappedComponent;
jest.mock("../../utils/get-config");
jest.mock("../../utils/validate-token");
jest.mock("../../utils/load-translation");
jest.mock("../../utils/submit-on-enter");
jest.mock("axios");

MobilePhoneChange.contextTypes = {
  setLoading: PropTypes.func,
  getLoading: PropTypes.func,
};

const createShallow = (props) =>
  shallow(<MobilePhoneChange {...props} />, {
    context: {setLoading: jest.fn(), getLoading: jest.fn()},
  });

const createTestProps = function (props, configName = "test-org-2") {
  const conf = getConfig(configName);
  const componentConf = conf.components.phone_number_change_form;
  componentConf.input_fields = {
    phone_number: conf.components.registration_form.input_fields.phone_number,
  };
  return {
    phone_number_change: componentConf,
    settings: conf.settings,
    orgSlug: conf.slug,
    orgName: conf.name,
    cookies: new Cookies(),
    logout: jest.fn(),
    setUserData: jest.fn(),
    userData: {},
    setTitle: jest.fn(),
    language: "en",
    // needed for subcomponents
    configuration: conf,
    ...props,
  };
};

describe("<MobilePhoneChange /> rendering with placeholder translation tags", () => {
  const props = createTestProps();
  it("should render translation placeholder correctly", () => {
    const wrapper = createShallow(props);
    expect(wrapper).toMatchSnapshot();
  });
});

const historyMock = createMemoryHistory();
historyMock.entries = [];
historyMock.location.key = "";

const mountComponent = function (props) {
  const mockedStore = {
    subscribe: () => {},
    dispatch: () => {},
    // needed to render <Contact/>
    getState: () => ({
      organization: {
        configuration: props.configuration,
      },
    }),
  };

  return mount(
    <Provider store={mockedStore}>
      <Router history={historyMock}>
        <MobilePhoneChangeWrapped {...props} />
      </Router>
    </Provider>,
    {
      context: {
        store: mockedStore,
        ...loadingContextValue,
      },
      childContextTypes: {
        store: PropTypes.object.isRequired,
        setLoading: PropTypes.func,
        getLoading: PropTypes.func,
      },
    },
  );
};

const userData = {
  response_code: "AUTH_TOKEN_VALIDATION_SUCCESSFUL",
  radius_user_token: "o6AQLY0aQjD3yuihRKLknTn8krcQwuy2Av6MCsFB",
  username: "tester@tester.com",
  is_active: false,
  phone_number: "+393660011222",
};

describe("Change Phone Number: standard flow", () => {
  let props;
  let wrapper;
  let lastConsoleOutuput;
  let originalError;
  const event = {preventDefault: jest.fn()};

  beforeEach(() => {
    props = createTestProps();
    validateToken.mockClear();
    // console mocking
    originalError = console.error;
    lastConsoleOutuput = null;
    console.error = (data) => {
      lastConsoleOutuput = data;
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
    jest.restoreAllMocks();
    console.error = originalError;
  });

  it("should render successfully", async () => {
    validateToken.mockReturnValue(true);
    props.userData = userData;
    loadTranslation("en", "default");
    wrapper = await mountComponent(props);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists(MobilePhoneChange)).toBe(true);
    expect(wrapper.find("input[name='phone_number']").length).toBe(1);

    const component = wrapper.find(MobilePhoneChange);
    expect(component.instance().state.phone_number).toBe("+393660011222");
    expect(component.find("form")).toHaveLength(1);
    expect(component.exists("#phone-number")).toBe(true);
    expect(component.exists("form input[type='submit']")).toBe(true);
    expect(component.exists(".row .button")).toBe(true);
  });

  it("should change phone number successfully", async () => {
    jest.spyOn(MobilePhoneChange.prototype, "handleSubmit");
    validateToken.mockReturnValue(true);
    jest.spyOn(toast, "info");
    jest.spyOn(historyMock, "push");
    axios.mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        statusText: "OK",
        data: null,
      }),
    );

    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    component.find("#phone-number").simulate("change", {
      target: {value: "+393660011333", name: "phone_number"},
    });
    expect(component.instance().state.phone_number).toBe("+393660011333");

    wrapper.find("form").simulate("submit", event);
    await tick();
    expect(event.preventDefault).toHaveBeenCalled();
    expect(MobilePhoneChange.prototype.handleSubmit).toHaveBeenCalled();
    expect(toast.info.mock.calls.length).toBe(1);
    expect(historyMock.push).toHaveBeenCalledWith(
      "/test-org-2/mobile-phone-verification",
    );
    expect(lastConsoleOutuput).toBe(null);
    const setUserDataMock = component.instance().props.setUserData.mock;
    expect(setUserDataMock.calls.length).toBe(1);
    expect(setUserDataMock.calls.pop()).toEqual([
      {is_verified: false, phone_number: "+393660011333"},
    ]);
  });
  it("should render PhoneInput lazily and handlers should work correctly", async () => {
    wrapper = createShallow(props);
    const handleChange = jest.spyOn(wrapper.instance(), "handleChange");
    const component = wrapper.find("Suspense");
    expect(component).toMatchSnapshot();
    expect(component.find("lazy").length).toBe(1);
    const prop = component.find("lazy").props();
    expect(prop).toEqual({
      enableSearch: false,
      excludeCountries: [],
      inputProps: {
        className: "form-control input ",
        id: "phone-number",
        name: "phone_number",
        required: true,
      },
      name: "phone_number",
      onChange: expect.any(Function),
      onKeyDown: expect.any(Function),
      onlyCountries: [],
      placeholder: "enter mobile phone number",
      preferredCountries: [],
      value: "",
    });
    prop.onChange("+911234567890");
    expect(handleChange).toHaveBeenCalledWith({
      target: {
        name: "phone_number",
        value: "++911234567890",
      },
    });
    component.find("lazy").props().onKeyDown({});
    expect(submitOnEnter.mock.calls.length).toEqual(1);
    expect(submitOnEnter.mock.calls.pop()).toEqual([
      {},
      expect.any(Object),
      "mobile-phone-change-form",
    ]);
  });

  it("should load fallback before PhoneInput and handlers should work correctly", async () => {
    wrapper = createShallow(props);
    const handleChange = jest.spyOn(wrapper.instance(), "handleChange");
    const component = wrapper.find("Suspense");
    const {fallback} = component.props();
    expect(fallback.type).toEqual("input");
    expect(fallback.props).toEqual({
      name: "phone_number",
      className: "form-control input",
      value: "",
      onChange: expect.any(Function),
      onKeyDown: expect.any(Function),
      placeholder: "enter mobile phone number",
      id: "phone-number",
    });
    fallback.props.onChange("+911234567890");
    expect(handleChange).toHaveBeenCalledWith({
      target: {
        name: "phone_number",
        value: "++911234567890",
      },
    });
    fallback.props.onKeyDown({});
    expect(submitOnEnter.mock.calls.length).toEqual(1);
    expect(submitOnEnter.mock.calls.pop()).toEqual([
      {},
      expect.any(Object),
      "mobile-phone-change-form",
    ]);
  });

  it("should render field error", async () => {
    jest.spyOn(MobilePhoneChange.prototype, "handleSubmit");
    jest.spyOn(toast, "info");
    jest.spyOn(historyMock, "push");
    axios.mockImplementationOnce(() =>
      Promise.reject({
        response: {
          status: 400,
          statusText: "OK",
          data: {
            phone_number: [
              "The new phone number must be different than the old one.",
            ],
          },
        },
      }),
    );

    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    wrapper.find("form").simulate("submit", event);
    await tick();
    expect(toast.info).not.toHaveBeenCalled();
    expect(historyMock.push).not.toHaveBeenCalled();
    expect(component.instance().state.errors.phone_number).toEqual([
      "The new phone number must be different than the old one.",
    ]);
    expect(component.instance().state.errors.nonField).toBeFalsy();
  });

  it("should render nonField error", async () => {
    jest.spyOn(MobilePhoneChange.prototype, "handleSubmit");
    jest.spyOn(toast, "info");
    jest.spyOn(historyMock, "push");
    axios.mockImplementationOnce(() =>
      Promise.reject({
        response: {
          status: 400,
          statusText: "OK",
          data: {
            non_field_errors: ["Maximum daily limit reached."],
          },
        },
      }),
    );

    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    wrapper.find("form").simulate("submit", event);
    await tick();
    expect(toast.info).not.toHaveBeenCalled();
    expect(historyMock.push).not.toHaveBeenCalled();
    expect(component.instance().state.errors.nonField).toEqual(
      "Maximum daily limit reached.",
    );
    expect(component.instance().state.errors.phone_number).toBeFalsy();
    expect(lastConsoleOutuput).not.toBe(null);
  });

  it("should cancel successfully", async () => {
    jest.spyOn(MobilePhoneChange.prototype, "handleSubmit");
    jest.spyOn(toast, "info");
    jest.spyOn(historyMock, "push");
    axios.mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        statusText: "OK",
        data: null,
      }),
    );

    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    const cancelButton = component.find(".cancel Link.button");
    cancelButton.simulate("click");
    expect(toast.info).not.toHaveBeenCalled();
    expect(historyMock.push).not.toHaveBeenCalled();
    expect(lastConsoleOutuput).toBe(null);
  });

  it("should set title", async () => {
    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    const setTitleMock = component.props().setTitle.mock;
    expect(setTitleMock.calls.pop()).toEqual([
      "Change mobile number",
      props.orgName,
    ]);
  });
});

describe("Change Phone Number: corner cases", () => {
  let props;
  let wrapper;
  const mockAxios = (responseData = {}) => {
    axios.mockImplementationOnce(() =>
      Promise.resolve({
        status: 200,
        statusText: "OK",
        data: {
          response_code: "AUTH_TOKEN_VALIDATION_SUCCESSFUL",
          radius_user_token: "o6AQLY0aQjD3yuihRKLknTn8krcQwuy2Av6MCsFB",
          username: "tester@tester.com",
          is_active: false,
          phone_number: "+393660011222",
          ...responseData,
        },
      }),
    );
  };

  beforeEach(() => {
    props = createTestProps();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.resetAllMocks();
    jest.restoreAllMocks();
  });

  it("should recognize if user is active", async () => {
    validateToken.mockReturnValue(true);
    userData.is_active = true;
    props.userData = userData;
    wrapper = await mountComponent(props);
    const component = wrapper.find(MobilePhoneChange);
    expect(component.instance().state.phone_number).toBe("+393660011222");
    const {setUserData} = component.instance().props;
    expect(setUserData.mock.calls.length).toBe(0);
  });

  it("should not redirect if mobile_phone_verification is enabled", async () => {
    mockAxios();
    props.settings.mobile_phone_verification = true;
    wrapper = await mountComponent(props);
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });

  it("should redirect if mobile_phone_verification disabled", async () => {
    props.settings.mobile_phone_verification = false;
    wrapper = await mountComponent(props);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });

  it("shouldn't redirect if user is active and mobile verificaton is true", async () => {
    validateToken.mockReturnValue(true);
    userData.is_active = true;
    props.userData = userData;
    props.settings.mobile_phone_verification = true;
    wrapper = await mountComponent(props);
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });

  it("should not redirect if user registration method is mobile_phone", async () => {
    validateToken.mockReturnValue(true);
    props.userData = userData;
    props.userData.is_active = true;
    props.userData.method = "mobile_phone";
    props.settings.mobile_phone_verification = true;
    wrapper = await mountComponent(props);
    expect(wrapper.find(Redirect)).toHaveLength(0);
  });

  it("should redirect if user registration method is not mobile_phone", async () => {
    validateToken.mockReturnValue(true);
    props.userData = userData;
    props.userData.is_active = true;
    props.userData.method = "saml";
    props.settings.mobile_phone_verification = true;
    wrapper = await mountComponent(props);
    expect(wrapper.find(Redirect)).toHaveLength(1);
  });

  it("should validate token", async () => {
    wrapper = await mountComponent(props);
    expect(validateToken).toHaveBeenCalledWith(
      props.cookies,
      props.orgSlug,
      props.setUserData,
      props.userData,
      props.logout,
      props.language,
    );
  });
});
