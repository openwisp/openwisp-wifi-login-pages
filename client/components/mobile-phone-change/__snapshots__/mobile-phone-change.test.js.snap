// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobilePhoneChange /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<div
  className="container content"
  id="mobile-phone-change"
>
  <div
    className="inner"
  >
    <form
      className="main-column"
      id="mobile-phone-change-form"
      onSubmit={[Function]}
    >
      <div
        className="inner"
      >
        <div
          className="row phone-number"
        >
          <label
            htmlFor="phone-number"
          >
            PHONE_LBL
          </label>
          <Suspense
            fallback={
              <input
                className="form-control input"
                id="phone-number"
                name="phone_number"
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="PHONE_PHOLD"
                value=""
              />
            }
          >
            <lazy
              enableSearch={false}
              excludeCountries={Array []}
              inputProps={
                Object {
                  "className": "form-control input ",
                  "id": "phone-number",
                  "name": "phone_number",
                  "required": true,
                }
              }
              name="phone_number"
              onChange={[Function]}
              onKeyDown={[Function]}
              onlyCountries={Array []}
              placeholder="PHONE_PHOLD"
              preferredCountries={Array []}
              value=""
            />
          </Suspense>
        </div>
        <div
          className="row submit"
        >
          <input
            className="button full"
            type="submit"
            value="PHONE_CHANGE_BTN"
          />
        </div>
        <div
          className="row cancel"
        >
          <Link
            className="button full"
            to="/test-org-2/status"
          >
            CANCEL
          </Link>
        </div>
      </div>
    </form>
    <Connect(Contact) />
  </div>
</div>
`;

exports[`Change Phone Number: standard flow should render PhoneInput lazily and handlers should work correctly 1`] = `
<Suspense
  fallback={
    <input
      className="form-control input"
      id="phone-number"
      name="phone_number"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="enter mobile phone number"
      value=""
    />
  }
>
  <lazy
    enableSearch={false}
    excludeCountries={Array []}
    inputProps={
      Object {
        "className": "form-control input ",
        "id": "phone-number",
        "name": "phone_number",
        "required": true,
      }
    }
    name="phone_number"
    onChange={[Function]}
    onKeyDown={[Function]}
    onlyCountries={Array []}
    placeholder="enter mobile phone number"
    preferredCountries={Array []}
    value=""
  />
</Suspense>
`;

exports[`Change Phone Number: standard flow should render successfully 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(MobilePhoneChange)
      configuration={
        Object {
          "auto_login": true,
          "components": Object {
            "contact_page": Object {
              "email": "support@openwisp.co",
              "helpdesk": "+789 948 564",
              "social_links": Array [
                Object {
                  "alt": "twitter",
                  "icon": "twitter.svg",
                  "url": "https://twitter.com/openwisp",
                },
                Object {
                  "alt": "facebook",
                  "icon": "facebook.svg",
                  "url": "https://facebook.com/openwisp",
                },
              ],
            },
            "mobile_phone_verification_form": Object {
              "input_fields": Object {
                "code": Object {
                  "pattern": "[0-9]{6,}",
                  "type": "text",
                },
              },
            },
            "phone_number_change_form": Object {
              "buttons": Object {
                "cancel": true,
                "change_phone_number": true,
              },
              "input_fields": Object {
                "phone_number": Object {
                  "country": "it",
                  "type": "tel",
                },
              },
            },
            "registration_form": Object {
              "additional_info_text": true,
              "input_fields": Object {
                "birth_date": Object {
                  "setting": "disabled",
                  "type": "date",
                },
                "city": Object {
                  "type": "text",
                },
                "country": Object {
                  "type": "text",
                },
                "email": Object {
                  "pattern": ".+@.+\\\\..+",
                  "type": "email",
                },
                "first_name": Object {
                  "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                  "setting": "disabled",
                  "type": "text",
                },
                "last_name": Object {
                  "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                  "setting": "disabled",
                  "type": "text",
                },
                "location": Object {
                  "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                  "setting": "disabled",
                  "type": "text",
                },
                "password": Object {
                  "pattern": ".{6,}",
                  "type": "password",
                },
                "password_confirm": Object {
                  "pattern": null,
                  "type": "password",
                },
                "phone_number": Object {
                  "country": "it",
                  "type": "tel",
                },
                "street": Object {
                  "type": "text",
                },
                "tax_number": Object {
                  "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                  "type": "text",
                },
                "username": Object {
                  "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                  "type": "text",
                },
                "zipcode": Object {
                  "type": "number",
                },
              },
              "links": Object {
                "login": true,
              },
            },
          },
          "default_language": "en",
          "name": "test org 2 (sms)",
          "privacy_policy": Object {
            "en": "<h1 id=\\"openwisp-wifi-login-pages-privacy-policy\\">openwisp-wifi-login-pages Privacy Policy</h1>
<ul>
<li><p>Your privacy is important to us. It is Openwisp's policy to respect your privacy regarding any information we may collect from you across openwisp-wifi-login-pages.</p>
</li>
<li><p>We only ask for personal information when we truly need it to provide a service to you. We collect it by fair and lawful means, with your knowledge and consent. We also let you know why we’re collecting it and how it will be used.</p>
</li>
<li><p>We only retain collected information for as long as necessary to provide you with your requested service. What data we store, we’ll protect within commercially acceptable means to prevent loss and theft, as well as unauthorised access, disclosure, copying, use or modification.</p>
</li>
<li><p>We don’t share any personally identifying information publicly or with third-parties, except when required to by law.</p>
</li>
<li><p>Our website may link to external sites that are not operated by us. Please be aware that we have no control over the content and practices of these sites, and cannot accept responsibility or liability for their respective privacy policies.</p>
</li>
<li><p>You are free to refuse our request for your personal information, with the understanding that we may be unable to provide you with some of your desired services.</p>
</li>
<li><p>Your continued use of our website will be regarded as acceptance of our practices around privacy and personal information. If you have any questions about how we handle user data and personal information, feel free to contact us.</p>
</li>
</ul>
",
          },
          "settings": Object {
            "mobile_phone_verification": true,
          },
          "slug": "test-org-2",
          "terms_and_conditions": Object {
            "en": "<h1 id=\\"openwisp-wifi-login-pages-terms-and-conditions\\">Openwisp-wifi-login-pages Terms and Conditions</h1>
<ul>
<li><p>By using or visiting the Openwisp-wifi-login-pages, you signify your agreement
to these terms and conditions (the \\"Terms of Service\\") and the Privacy policy.</p>
</li>
<li><p>In order to access the wifi service, you need to be a registered user of the
default organization.</p>
</li>
<li><p>Openwisp-wifi-login-pages is not liable for any unauthorized use of your account,
you may be liable for the losses of the services or others due to such unauthorized use.</p>
</li>
<li><p>Openwisp reserves the right to discontinue or disallow any user, group or any
specific IP (internal or external) from using its services without any prior notice.</p>
</li>
<li><p>Violating the above mentioned Terms and Conditions may result in the termination
of the particular user account or the particular IP address from accessing
Openwisp-wifi-login-pages and using its services in the future.</p>
</li>
</ul>
",
          },
        }
      }
      cookies={
        Cookies {
          "HAS_DOCUMENT_COOKIE": true,
          "changeListeners": Array [],
          "cookies": Object {},
        }
      }
      language="en"
      logout={[MockFunction]}
      orgName="test org 2 (sms)"
      orgSlug="test-org-2"
      phone_number_change={
        Object {
          "buttons": Object {
            "cancel": true,
            "change_phone_number": true,
          },
          "input_fields": Object {
            "phone_number": Object {
              "country": "it",
              "type": "tel",
            },
          },
        }
      }
      setTitle={
        [MockFunction] {
          "calls": Array [
            Array [
              "Change mobile number",
              "test org 2 (sms)",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      setUserData={[MockFunction]}
      settings={
        Object {
          "mobile_phone_verification": true,
        }
      }
      userData={
        Object {
          "is_active": false,
          "phone_number": "+393660011222",
          "radius_user_token": "o6AQLY0aQjD3yuihRKLknTn8krcQwuy2Av6MCsFB",
          "response_code": "AUTH_TOKEN_VALIDATION_SUCCESSFUL",
          "username": "tester@tester.com",
        }
      }
    >
      <MobilePhoneChange
        configuration={
          Object {
            "auto_login": true,
            "components": Object {
              "contact_page": Object {
                "email": "support@openwisp.co",
                "helpdesk": "+789 948 564",
                "social_links": Array [
                  Object {
                    "alt": "twitter",
                    "icon": "twitter.svg",
                    "url": "https://twitter.com/openwisp",
                  },
                  Object {
                    "alt": "facebook",
                    "icon": "facebook.svg",
                    "url": "https://facebook.com/openwisp",
                  },
                ],
              },
              "mobile_phone_verification_form": Object {
                "input_fields": Object {
                  "code": Object {
                    "pattern": "[0-9]{6,}",
                    "type": "text",
                  },
                },
              },
              "phone_number_change_form": Object {
                "buttons": Object {
                  "cancel": true,
                  "change_phone_number": true,
                },
                "input_fields": Object {
                  "phone_number": Object {
                    "country": "it",
                    "type": "tel",
                  },
                },
              },
              "registration_form": Object {
                "additional_info_text": true,
                "input_fields": Object {
                  "birth_date": Object {
                    "setting": "disabled",
                    "type": "date",
                  },
                  "city": Object {
                    "type": "text",
                  },
                  "country": Object {
                    "type": "text",
                  },
                  "email": Object {
                    "pattern": ".+@.+\\\\..+",
                    "type": "email",
                  },
                  "first_name": Object {
                    "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                    "setting": "disabled",
                    "type": "text",
                  },
                  "last_name": Object {
                    "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                    "setting": "disabled",
                    "type": "text",
                  },
                  "location": Object {
                    "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                    "setting": "disabled",
                    "type": "text",
                  },
                  "password": Object {
                    "pattern": ".{6,}",
                    "type": "password",
                  },
                  "password_confirm": Object {
                    "pattern": null,
                    "type": "password",
                  },
                  "phone_number": Object {
                    "country": "it",
                    "type": "tel",
                  },
                  "street": Object {
                    "type": "text",
                  },
                  "tax_number": Object {
                    "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                    "type": "text",
                  },
                  "username": Object {
                    "pattern": "[a-zA-Z@.+\\\\-_\\\\d]{1,150}",
                    "type": "text",
                  },
                  "zipcode": Object {
                    "type": "number",
                  },
                },
                "links": Object {
                  "login": true,
                },
              },
            },
            "default_language": "en",
            "name": "test org 2 (sms)",
            "privacy_policy": Object {
              "en": "<h1 id=\\"openwisp-wifi-login-pages-privacy-policy\\">openwisp-wifi-login-pages Privacy Policy</h1>
<ul>
<li><p>Your privacy is important to us. It is Openwisp's policy to respect your privacy regarding any information we may collect from you across openwisp-wifi-login-pages.</p>
</li>
<li><p>We only ask for personal information when we truly need it to provide a service to you. We collect it by fair and lawful means, with your knowledge and consent. We also let you know why we’re collecting it and how it will be used.</p>
</li>
<li><p>We only retain collected information for as long as necessary to provide you with your requested service. What data we store, we’ll protect within commercially acceptable means to prevent loss and theft, as well as unauthorised access, disclosure, copying, use or modification.</p>
</li>
<li><p>We don’t share any personally identifying information publicly or with third-parties, except when required to by law.</p>
</li>
<li><p>Our website may link to external sites that are not operated by us. Please be aware that we have no control over the content and practices of these sites, and cannot accept responsibility or liability for their respective privacy policies.</p>
</li>
<li><p>You are free to refuse our request for your personal information, with the understanding that we may be unable to provide you with some of your desired services.</p>
</li>
<li><p>Your continued use of our website will be regarded as acceptance of our practices around privacy and personal information. If you have any questions about how we handle user data and personal information, feel free to contact us.</p>
</li>
</ul>
",
            },
            "settings": Object {
              "mobile_phone_verification": true,
            },
            "slug": "test-org-2",
            "terms_and_conditions": Object {
              "en": "<h1 id=\\"openwisp-wifi-login-pages-terms-and-conditions\\">Openwisp-wifi-login-pages Terms and Conditions</h1>
<ul>
<li><p>By using or visiting the Openwisp-wifi-login-pages, you signify your agreement
to these terms and conditions (the \\"Terms of Service\\") and the Privacy policy.</p>
</li>
<li><p>In order to access the wifi service, you need to be a registered user of the
default organization.</p>
</li>
<li><p>Openwisp-wifi-login-pages is not liable for any unauthorized use of your account,
you may be liable for the losses of the services or others due to such unauthorized use.</p>
</li>
<li><p>Openwisp reserves the right to discontinue or disallow any user, group or any
specific IP (internal or external) from using its services without any prior notice.</p>
</li>
<li><p>Violating the above mentioned Terms and Conditions may result in the termination
of the particular user account or the particular IP address from accessing
Openwisp-wifi-login-pages and using its services in the future.</p>
</li>
</ul>
",
            },
          }
        }
        cookies={
          Cookies {
            "HAS_DOCUMENT_COOKIE": true,
            "changeListeners": Array [],
            "cookies": Object {},
          }
        }
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        language="en"
        location={
          Object {
            "hash": "",
            "key": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        logout={[MockFunction]}
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        orgName="test org 2 (sms)"
        orgSlug="test-org-2"
        phone_number_change={
          Object {
            "buttons": Object {
              "cancel": true,
              "change_phone_number": true,
            },
            "input_fields": Object {
              "phone_number": Object {
                "country": "it",
                "type": "tel",
              },
            },
          }
        }
        setTitle={
          [MockFunction] {
            "calls": Array [
              Array [
                "Change mobile number",
                "test org 2 (sms)",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        setUserData={[MockFunction]}
        settings={
          Object {
            "mobile_phone_verification": true,
          }
        }
        userData={
          Object {
            "is_active": false,
            "phone_number": "+393660011222",
            "radius_user_token": "o6AQLY0aQjD3yuihRKLknTn8krcQwuy2Av6MCsFB",
            "response_code": "AUTH_TOKEN_VALIDATION_SUCCESSFUL",
            "username": "tester@tester.com",
          }
        }
      >
        <div
          className="container content"
          id="mobile-phone-change"
        >
          <div
            className="inner"
          >
            <form
              className="main-column"
              id="mobile-phone-change-form"
              onSubmit={[Function]}
            >
              <div
                className="inner"
              >
                <div
                  className="row phone-number"
                >
                  <label
                    htmlFor="phone-number"
                  >
                    Mobile phone number
                  </label>
                  <Suspense
                    fallback={
                      <input
                        className="form-control input"
                        id="phone-number"
                        name="phone_number"
                        onChange={[Function]}
                        onKeyDown={[Function]}
                        placeholder="enter mobile phone number"
                        value=""
                      />
                    }
                  >
                    <input
                      className="form-control input"
                      id="phone-number"
                      name="phone_number"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="enter mobile phone number"
                      value=""
                    />
                  </Suspense>
                </div>
                <div
                  className="row submit"
                >
                  <input
                    className="button full"
                    type="submit"
                    value="Change your phone number"
                  />
                </div>
                <div
                  className="row cancel"
                >
                  <Link
                    className="button full"
                    to="/test-org-2/status"
                  >
                    <LinkAnchor
                      className="button full"
                      href="/test-org-2/status"
                      navigate={[Function]}
                    >
                      <a
                        className="button full"
                        href="/test-org-2/status"
                        onClick={[Function]}
                      >
                        Cancel
                      </a>
                    </LinkAnchor>
                  </Link>
                </div>
              </div>
            </form>
            <Connect(Contact)>
              <Contact
                contactPage={
                  Object {
                    "email": "support@openwisp.co",
                    "helpdesk": "+789 948 564",
                    "social_links": Array [
                      Object {
                        "alt": "twitter",
                        "icon": "twitter.svg",
                        "url": "https://twitter.com/openwisp",
                      },
                      Object {
                        "alt": "facebook",
                        "icon": "facebook.svg",
                        "url": "https://facebook.com/openwisp",
                      },
                    ],
                  }
                }
                dispatch={[Function]}
                isAuthenticated={false}
                orgSlug="test-org-2"
                userData={Object {}}
              >
                <div
                  className="side-column contact"
                >
                  <div
                    className="inner"
                  >
                    <div
                      className="row"
                    >
                      <span
                        className="label"
                      >
                        Email
                        :
                      </span>
                      <a
                        className="link"
                        href="mailto:support@openwisp.co"
                      >
                        support@openwisp.co
                      </a>
                    </div>
                    <div
                      className="row"
                    >
                      <span
                        className="label"
                      >
                        Helpdesk
                        :
                      </span>
                      <a
                        className="link"
                        href="tel:+789 948 564"
                      >
                        +789 948 564
                      </a>
                    </div>
                    <div
                      className="contact-links"
                    >
                      <a
                        className="link "
                        href="https://twitter.com/openwisp"
                        key="https://twitter.com/openwisp"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <img
                          className="contact-image "
                          src="/assets/test-org-2/twitter.svg"
                        />
                      </a>
                      <a
                        className="link "
                        href="https://facebook.com/openwisp"
                        key="https://facebook.com/openwisp"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <img
                          className="contact-image "
                          src="/assets/test-org-2/facebook.svg"
                        />
                      </a>
                    </div>
                  </div>
                </div>
              </Contact>
            </Connect(Contact)>
          </div>
        </div>
      </MobilePhoneChange>
    </withRouter(MobilePhoneChange)>
  </Router>
</Provider>
`;
